test/hotspot/jtreg/runtime/cds/appcds/applications/JavacBenchApp.java

```diff
@@ -0,0 +1,228 @@
+                x = Long.parseLong(args[0]);
+            } catch (Throwable t) {
+                t.printStackTrace();
+            }
+            doit(() -> {
+                System.out.println("Hello Lambda");
+                Thread.dumpStack();
+            });
+        }
+        static List<String> list = List.of("1", "2");
+        class InnerClass1 {
+            static final long yy = y;
+        }
+        static void doit(Runnable r) {
+            for (var x : list) {
+                r.run();
+            }
+        }
+        static String patternMatch(String arg, Object o) {
+            if (o instanceof String s) {
+                return "1234";
+            }
+            final String b = "B";
+            return switch (arg) {
+                case "A" -> "a";
+                case b   -> "b";
+                default  -> "c";
+            };
+        }
+        public sealed class SealedInnerClass {}
+        public final class Foo extends SealedInnerClass {}
+        enum Expression {
+            ADDITION,
+            SUBTRACTION,
+            MULTIPLICATION,
+            DIVISION
+        }
+        public record Point(int x, int y) {
+            public Point(int x) {
+                this(x, 0);
+            }
+        }
+        """;
+
+    String sanitySource = """
+        public class Sanity implements java.util.concurrent.Callable<String> {
+            public String call() {
+                return "this is a test";
+            }
+        }
+        """;
+
+    void setup(int count) {
+        sources = new ArrayList<>(count);
+        for (int i = 0; i < count; i++) {
+            String source = imports + "public class Test" + i + " {" + testClassBody + "}";
+            sources.add(new SourceFile("Test" + i, source));
+        }
+
+        sources.add(new SourceFile("Sanity", sanitySource));
+    }
+
+    @SuppressWarnings("unchecked")
+    static void validate(byte[] sanityClassFile) throws Throwable {
+        MethodHandles.Lookup lookup = MethodHandles.lookup();
+        Class<?> cls = lookup.defineClass(sanityClassFile);
+        Callable<String> obj = (Callable<String>)cls.getDeclaredConstructor().newInstance();
+        String s = obj.call();
+        if (!s.equals("this is a test")) {
+            throw new RuntimeException("Expected \"this is a test\", but got \"" + s + "\"");
+        }
+    }
+
+    public static void main(String args[]) throws Throwable {
+        long started = System.currentTimeMillis();
+        JavacBenchApp bench = new JavacBenchApp();
+
+        int count = 0;
+        if (args.length > 0) {
+            count = Integer.parseInt(args[0]);
+            if (count >= 0) {
+                bench.setup(count);
+                Map<String, byte[]> allClasses = bench.compile();
+                validate(allClasses.get("Sanity"));
+            }
+        }
+        long elapsed = System.currentTimeMillis() - started;
+        if (System.getProperty("JavacBenchApp.silent") == null) {
```

Comment:
Should line 221 be inside the 'if' block?

Reply:
I moved it inside the 'if' block.